.github/scripts/main/main.sh

```diff
@@ -52,6 +52,48 @@ make install
 export PATH="$PREFIX/bin:$PATH"
 opam --version
 
+if [[ $OPAM_DOC -eq 1 ]]; then
+  make -C doc html man-html pages
+
+  if [ "$GITHUB_EVENT_NAME" = "pull_request" ]; then
+    . .github/scripts/common/hygiene-preamble.sh
+    diff="git diff $BASE_REF_SHA..$PR_REF_SHA"
+    set +e
+    files="$($diff --name-only --diff-filter=A -- src/**/*.mli)"
+    set -e
+    if [ -n "$files" ]; then
+      echo '::group::new module(s) added - check index updated too'
+      if $diff --name-only --exit-code -- doc/index.html ; then
+        echo '::error new module(s) added but doc/index.html not updated'
+        echo "$files"
+        exit 3
+      fi
+      echo '::endgroup::new module added - checking it'
+    else
+      echo 'No new modules added'
+    fi
+  fi
+
+  htmlfiles=$(ls doc/pages/*.md | sed -e 's/\.md/.html/')
+  manfiles=$(opam --help=groff | grep '^\\fB\w' | sed 's/\\fB\(.*\)\\fR .*/doc\/man-html\/opam-\1.html/' | grep -v 'opam-opam')
+
+  echo '::group::checking for generated files'
+  echo "htmlfiles: $htmlfiles"
+  echo "manfiles: $manfiles"
```

Comment:
Just for slightly clearer explanation of what they are (than the variable names - only because the one called manfiles also lists .html files... it makes sense in the code, but less so in the log)
```suggestion
  echo "pages: $htmlfiles"
  echo "topics: $manfiles"
```